dist/rules/no-internal.js

```diff
@@ -111,48 +112,60 @@ module.exports = {
     }
 
     /**
-     * Checks if the package that owns the specified file path matches a checked package pattern regex.s
+     * Returns true if the internal tag is a violation within the package that owns the specified file.
      * @param {string} filePath
      */
     function owningPackageIsCheckedPackage(filePath) {
-      const packageList = workspace.getWorkspaces(filePath);
+      let owningPackageJson;
+      let lintedPackageJson;
       
-      // Look through all package infos to find the one containing our packagePath
-      let packageObj = packageList.find((pkg) => {
-        const packageBaseDir = path.dirname(pkg.packageJson.packageJsonPath);
-        return dirContainsPath(packageBaseDir, filePath);
-      });
+      // Find package.json of the file being linted
+      const lintedPackageJsonPath = findup("package.json", { cwd: context.filename });
+      if (lintedPackageJsonPath !== null) {
+        lintedPackageJson = JSON.parse(readFileSync(lintedPackageJsonPath, "utf8"));
+      }
+
+      // Find package.json of the owning package
+      const owningPackageJsonPath = findup("package.json", { cwd: filePath });
+      if (owningPackageJsonPath !== null) {
+        owningPackageJson = JSON.parse(readFileSync(owningPackageJsonPath, "utf8"));
+      }
+
+      // If both packages are in the itwinjs-core repository, allow internal tags
+      if (
+        typeof owningPackageJson?.repository !== "string" &&
+        owningPackageJson?.repository?.url === "https://github.com/iTwin/itwinjs-core.git" &&
```

Comment:
idk if we want to expose the itwinjs-internal repos url, but we do want to allow that repo or rather some pkgs from that repo to use internal apis. wonder if we should rather add an additional field in the pkg.json to denote its an internal pkg, and its safe to use internal apis.

Reply:
Is this field added in the consumers of itwin core? Wouldn't we want to control who is allowed to use internal apis

Reply:
Maybe I'm misunderstanding, but we don't want iTwin to use internal APIs from _outside_ of iTwin, so I don't see how a setting in package.json would help, unless we look for said flag and then check to see if the URL from both lintedPackageJson and owningPackageJson are the same.

Reply:
> " we don't want iTwin to use internal APIs from outside of iTwin"

No, we don't want pkgs outside of itwinjs-core (and a select few pkgs) to be using internal apis defined by itwinjs-core.

Reply:
Should this discussion result in a possible separate PR? Or does it need to be resolved in this PR?

Reply:
Yea, im ok handling this in a separate pr

Reply:
created issue, https://github.com/iTwin/eslint-plugin/issues/80